File: front/src/app/Services/Reports/report.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environemnt } from '../../../environment/Environment';
import { AttendanceReport, AttendanceReportRequest, EarningCongressFilter, EarningCongressReport, EarningFilter, EarningReport, InscriptionFilter, InscriptionReport, WorkshopReport, WorkshopReportFilter } from '../../interfaces/Report';
import { Observable } from 'rxjs';
import { CongressResponse } from '../../interfaces/Congress';

@Injectable({
  providedIn: 'root'
})
export class ReportService {
  baseUrl = environemnt.baseUrl + '/reports';

  constructor(private http: HttpClient) { }

  getEarningsReport(data: EarningFilter): Observable<EarningReport[]> {
    let params = new HttpParams()
      .set('startDate', data.startDate.substring(0, 10))  
      .set('endDate', data.endDate.substring(0, 10));
    if (data.organizationId !== null && data.organizationId !== undefined)
      params = params.set('organizationId', data.organizationId);
    return this.http.get<EarningReport[]>(`${this.baseUrl}/earnings`, { params });
  }

  getCongressByOrganization(organizationId: number): Observable<CongressResponse[]> {
    return this.http.get<CongressResponse[]>(`${this.baseUrl}/organizations/${organizationId}/congress`);
  }

  getInscriptionReport(filter: InscriptionFilter): Observable<InscriptionReport[]> {
    return this.http.post<InscriptionReport[]>(`${this.baseUrl}/participants`, filter);
  }

  getAttendanceReport(request: AttendanceReportRequest): Observable<AttendanceReport[]> {
    return this.http.post<AttendanceReport[]>(`${this.baseUrl}/attendance`, request);
  }

  getWorkshopReport(filter: WorkshopReportFilter): Observable<WorkshopReport[]> {
    return this.http.post<WorkshopReport[]>(`${this.baseUrl}/workshops`, filter);
  }

  getEarningsCongressReport(filter: EarningCongressFilter): Observable<EarningCongressReport[]> {
    return this.http.post<EarningCongressReport[]>(`${this.baseUrl}/congress/earnings`, filter);
  }
}

File: back/src/main/java/com/ayd2/congress/services/reports/ReportService.java
package com.ayd2.congress.services.reports;

import java.util.List;

import com.ayd2.congress.dtos.Congress.CongressResponse;
import com.ayd2.congress.dtos.reports.AtteendanceReporRequest;
import com.ayd2.congress.dtos.reports.AtteendanceReport;
import com.ayd2.congress.dtos.reports.EarningCongressReport;
import com.ayd2.congress.dtos.reports.EarningFilter;
import com.ayd2.congress.dtos.reports.EarningReport;
import com.ayd2.congress.dtos.reports.EarningCongressFilter;
import com.ayd2.congress.dtos.reports.InscriptionFilter;
import com.ayd2.congress.dtos.reports.InscriptionReport;
import com.ayd2.congress.dtos.reports.WorkshopReport;
import com.ayd2.congress.dtos.reports.WorkshopReportFilter;
import com.ayd2.congress.exceptions.NotFoundException;

public interface ReportService {
    //admin reports
   List<EarningReport> earningsReport(EarningFilter filter) throws NotFoundException;
    List<CongressResponse> congressByOrganizationId(Long id) throws NotFoundException;

    //congress admin reports
    List<InscriptionReport> inscriptionReport(InscriptionFilter filter) throws NotFoundException;
    List<AtteendanceReport> atteendanceReport(AtteendanceReporRequest reportRequest) throws NotFoundException;
    List<WorkshopReport> atteendanceWorkshop(WorkshopReportFilter filter) throws NotFoundException;
    List<EarningCongressReport> earningsCongressReport(EarningCongressFilter filter) throws NotFoundException;

}
